src/MoBi.Core/Serialization/Services/SerializationContextFactory.cs

```diff
@@ -17,26 +21,50 @@ public class SerializationContextFactory : ISerializationContextFactory
       private readonly ISerializationDimensionFactory _dimensionFactory;
       private readonly IObjectBaseFactory _objectBaseFactory;
       private readonly ICloneManagerForModel _cloneManagerForModel;
+      private readonly IContainer _container;
 
-      public SerializationContextFactory(ISerializationDimensionFactory dimensionFactory, IObjectBaseFactory objectBaseFactory,
-         ICloneManagerForModel cloneManagerForModel)
+      public SerializationContextFactory(
+         ISerializationDimensionFactory dimensionFactory,
+         IObjectBaseFactory objectBaseFactory,
+         ICloneManagerForModel cloneManagerForModel,
+         IContainer container
+      )
       {
          _dimensionFactory = dimensionFactory;
          _objectBaseFactory = objectBaseFactory;
          _cloneManagerForModel = cloneManagerForModel;
+         _container = container;
       }
 
       public SerializationContext Create(SerializationContext parentSerializationContext = null)
       {
-         var serializationContext = SerializationTransaction.Create(_dimensionFactory, _objectBaseFactory, new WithIdRepository(), _cloneManagerForModel);
+         var projectRetriever = _container.Resolve<IMoBiProjectRetriever>();
+         var project = projectRetriever.Current;
+
+         var idRepository = new WithIdRepository();
+         var allRepositories = new List<DataRepository>();
 
          if (parentSerializationContext != null)
          {
-            parentSerializationContext.Repositories.Each(serializationContext.AddRepository);
-            parentSerializationContext.IdRepository.All().Each(serializationContext.Register);
+            allRepositories.AddRange(parentSerializationContext.Repositories);
+            parentSerializationContext.IdRepository.All().Each(idRepository.Register);
+         }
+         
+         //if project is defined, retrieved all available results from existing simulation. Required to ensure correct deserialization
+         if (project != null)
```

Comment:
This is the big change in MoBi. Similar to what PKSim is doing, we need to ensure that we load observed data and simulations when we are loading a related item such as PI